ui/src/components/EdgePins.tsx

```diff
@@ -0,0 +1,159 @@
+import L, { LatLng, LatLngBounds } from "leaflet";
+import * as React from "react";
+import "react-leaflet-markercluster/dist/styles.min.css";
+import "leaflet.markercluster";
+import "leaflet/dist/leaflet.css";
+import { EdgePinCluster } from "./EdgePinCluster";
+import { Story } from "../types";
+
+export enum EdgeType {
+  Bottom,
+  Right,
+  Top,
+  Left,
+}
+
+function getEdgeType(index) {
+  switch (index) {
+    case 0:
+      return EdgeType.Bottom;
+    case 1:
+      return EdgeType.Right;
+    case 2:
+      return EdgeType.Top;
+    default:
+      return EdgeType.Left;
+  }
+}
+
+function intersects(a, b, c, d, p, q, r, s) {
+  const det = (c - a) * (s - q) - (r - p) * (d - b);
+  if (det === 0) {
+    return false;
+  } else {
+    const lambda = ((s - q) * (r - a) + (p - r) * (s - b)) / det;
+    const gamma = ((b - d) * (r - a) + (c - a) * (s - b)) / det;
+    return 0 < lambda && lambda < 1 && 0 < gamma && gamma < 1;
+  }
+}
+
+function getPointOfIntersection(a, b, c, d, p, q, r, s) {
+  const det = (c - a) * (s - q) - (r - p) * (d - b);
+  const lambda = ((s - q) * (r - a) + (p - r) * (s - b)) / det;
+
+  return L.latLng(a + lambda * (c - a), b + lambda * (d - b));
+}
+
+function angleFromCoordinate(lat1, long1, lat2, long2) {
+  const dLon = long2 - long1;
+
+  const y = Math.sin(dLon) * Math.cos(lat2);
+  const x =
+    Math.cos(lat1) * Math.sin(lat2) -
+    Math.sin(lat1) * Math.cos(lat2) * Math.cos(dLon);
+
+  const brng = (Math.atan2(y, x) * 180) / Math.PI;
+
+  return brng - 80;
+}
+
+export interface EdgePinsProps {
+  stories: Story[];
+  currPosition: LatLng;
+  mapBounds: LatLngBounds;
+}
+
+export function EdgePins({
+  stories,
+  currPosition,
+  mapBounds,
+}: EdgePinsProps): JSX.Element {
+  // determine if the current view has markers
+  if (!mapBounds || !currPosition) return null;
+  let isVisible = true;
+
+  const pinPositions = stories.map((story) => {
+    return { lat: story.latitude, lng: story.longitude };
+  });
+
+  pinPositions.forEach((pos) => {
+    if (mapBounds.contains(L.latLng(pos.lat, pos.lng))) {
+      isVisible = false;
+    }
+  });
```

Comment:
Lets use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some here.

```suggestion
const isVisible = pinPositions.some((pos) => mapBounds.contains(L.latLng(pos.lat, pos.lng)));
```